File: zvault/web/js/tx/test_network_constants.mjs
/**
 * X1: testnet constants must never apply on mainnet.
 * Run: node js/tx/test_network_constants.mjs
 */
import {
  TREASURY_MAINNET,
  TREASURY_TESTNET,
  assertTreasuryForNetwork,
  constantsForNetwork,
} from "../config.js";

function assert(cond, msg) {
  if (!cond) throw new Error(msg || "assert");
}

function throws(fn, needle) {
  let err;
  try {
    fn();
  } catch (e) {
    err = e;
  }
  assert(err, `expected throw (${needle})`);
  assert(String(err.message).includes(needle), err.message);
}

assert(constantsForNetwork("main").treasury === TREASURY_MAINNET);
assert(constantsForNetwork("test").treasury === TREASURY_TESTNET);
assert(constantsForNetwork("test").treasury.startsWith("tm"));
assert(constantsForNetwork("main").treasury.startsWith("t1"));

throws(() => assertTreasuryForNetwork(TREASURY_TESTNET, "main"), "testnet treasury");
throws(() => assertTreasuryForNetwork(TREASURY_MAINNET, "test"), "mainnet treasury");
throws(() => constantsForNetwork(""), "network required");
throws(() => constantsForNetwork("bogus"), "unknown network");

assertTreasuryForNetwork(TREASURY_TESTNET, "test");
assertTreasuryForNetwork(TREASURY_MAINNET, "main");

console.log("X1 network constants OK — testnet treasury blocked on mainnet");

File: zvault/zcash/test_network_constants.py
#!/usr/bin/env python3
"""X1: testnet treasury must never be selected for mainnet."""
import sys
from pathlib import Path
sys.path.insert(0, str(Path(__file__).resolve().parent))
import indexer as ix

t_main, h_main = ix.constants_for_network("main")
t_test, h_test = ix.constants_for_network("test")
assert t_main == ix.TREASURY_MAINNET and t_main.startswith("t1")
assert t_test == ix.TREASURY_TESTNET and t_test.startswith("tm")
assert t_main != t_test

try:
    ix.assert_treasury_matches_network(ix.TREASURY_TESTNET, "main")
    raise SystemExit("expected refusal of testnet treasury on mainnet")
except ValueError as e:
    assert "testnet treasury" in str(e)

try:
    ix.assert_treasury_matches_network(ix.TREASURY_MAINNET, "test")
    raise SystemExit("expected refusal of mainnet treasury on testnet")
except ValueError as e:
    assert "mainnet treasury" in str(e)

ix.assert_treasury_matches_network(ix.TREASURY_TESTNET, "test")
ix.assert_treasury_matches_network(ix.TREASURY_MAINNET, "main")
print("X1 python network constants OK")
